python/rpdk/java/templates/init/guided_aws/TagHelper.java

```diff
@@ -0,0 +1,256 @@
+        return currentTags;
+    }
+
+    /**
+     * getNewDesiredTags
+     *
+     * If stack tags and resource tags are not merged together in Configuration class,
+     * we will get new user defined tags from both resource model and previous stack tags.
+     */
+    public Map<String, String> getNewDesiredTags(final ResourceModel resourceModel, final ResourceHandlerRequest<ResourceModel> handlerRequest) {
+        // get new stack level tags from handlerRequest
+        final Map<String, String> desiredTags = handlerRequest.getDesiredResourceTags();
+
+        // TODO: get tags from resource model based on your tag property name
+        // TODO: desiredTags.putAll(convertToMap(resourceModel.getTags()));
+        return desiredTags;
+    }
+
+    /**
+     * getTagsToAdd
+     *
+     * Determines the tags the customer desired to define or redefine.
+     */
+    public Map<String, String> generateTagsToAdd(final Map<String, String> currentTags, final Map<String, String> desiredTags) {
+        return desiredTags.entrySet().stream()
+            .filter(e -> !currentTags.containsKey(e.getKey()) || !Objects.equals(currentTags.get(e.getKey()), e.getValue()))
+            .collect(Collectors.toMap(
+                Map.Entry::getKey,
+                Map.Entry::getValue));
+    }
+
+    /**
+     * getTagsToRemove
+     *
+     * Determines the tags the customer desired to remove from the function.
+     */
+    public Set<String> generateTagsToRemove(final Map<String, String> currentTags, final Map<String, String> desiredTags) {
+        final Set<String> desiredTagNames = desiredTags.keySet();
+
+        return currentTags.keySet().stream()
+            .filter(tagName -> !desiredTagNames.contains(tagName))
+            .collect(Collectors.toSet());
+    }
+
+    /**
+     * getTagsToAdd
+     *
+     * Determines the tags the customer desired to define or redefine.
+     */
+    public Set<Tag> generateTagsToAdd(final Set<Tag> currentTags, final Set<Tag> desiredTags) {
+        return Sets.difference(new HashSet<>(desiredTags), new HashSet<>(currentTags));
+    }
+
+    /**
+     * getTagsToRemove
+     *
+     * Determines the tags the customer desired to remove from the function.
+     */
+    public Set<Tag> generateTagsToRemove(final Set<Tag> currentTags, final Set<Tag> desiredTags) {
+        return Sets.difference(new HashSet<>(currentTags), new HashSet<>(desiredTags));
+    }
+
+    /**
+     * doUpdateTags
+     *
+     * This function updates the target resource's user defined tags. It first
+     * calculate what tags to add and remove, and then calls TagResource/UntagResource
+     * to achieve the customer's desired set of tags as specified in their CloudFormation template.
+     *
+     */
+    public ProgressEvent<ResourceModel, CallbackContext> doUpdateTags(final AmazonWebServicesClientProxy proxy, final ProxyClient<SdkClient> serviceClient,
```

Comment:
I don't know how much useful of those methods, since most of them are uncompleted. The call chain framework is also one of the options and we should not too opinionated of how customer will implement their handlers.

Maybe it should start from small with provide some tags utils like the tags for add or remove.  We may can provide even better abstraction to help to remove and delete tags without those boilerplate code.

Reply:
make sense, removing this method

Reply:
moving this method